source/clear-linux/guides/network/ipxe-install.rst

```diff
@@ -0,0 +1,328 @@
+##############################################
+
+This guide shows how to install |CLOSIA| through a
+:abbr:`PXE (pre-boot execution environment)`. PXE is an industry standard
+describing the client-server interaction with network-boot software using
+the DHCP and TFTP protocols. This guide shows one possible use of this
+environment to automatically install |CL|.
+
+The PXE extension known as `iPXE`_\* adds support for additional protocols
+such as HTTP, iSCIS, :abbr:`AoE (ATA over Ethernet)`, and
+:abbr:`FCoE (Fiber Channel over Ethernet)`. iPXE can also be used to enable
+network booting on computers which lack built-in PXE support.
+
+Figure 1 depicts the flow of information between a PXE server and a PXE
+client we must create to install |CL| through iPXE.
+
+.. figure:: ./figures/network-boot-flow.png
+   :alt: PXE information flow
+
+   Figure 1: PXE information flow
+
+This guide covers how to install |CL| through iPXE.
+
+.. caution::
+
+   This process erases existing data and partitions on the PXE client
+   system. |CL| automatically erases all data on the PXE client and
+   creates 3 new partitions to install onto.
+
+Prerequisites
+*************
+
+Before booting with iPXE, the following preparations must be made:
+
+* Your PXE server has an Ethernet/LAN boot option.
+* Your PXE server has at least two network adapters.
+* Your PXE server is connected to a public network.
+* Your PXE server and PXE clients are connected to a switch on a private
+  network.
+* Your PXE server has the secure boot option disabled.
+* Your PXE clients have a boot order where the network boot option is
+  prioritized before the disk boot option.
+
+.. note::
+
+   The secure boot option must be disabled because the UEFI binaries used to
+   boot |CL| are not signed.
+
+The required computer and network setup is shown in figure 2.
+
+.. figure:: ./figures/network-boot-setup.png
+   :alt: NAT network topology
+
+   Figure 2: NAT network topology
+
+Configuration
+*************
+
+The configuration process to install |CL| using iPXE has been automated with
+the :file:`configure-ipxe.sh` script included with
+:abbr:`ICIS (Ister Cloud Init Service)`, thus quickly enabling a bulk
+provisioning setup. For additional instructions on how to get started with the
+script, refer to the guide on the `ICIS GitHub repository`_. Otherwise, to
+setup manually, follow the steps below.
+
+#. Define the variables used to parameterize the configuration of an iPXE
+   boot with the following commands:
+
+   .. code-block:: console
+
+      ipxe_app_name=ipxe
+      ipxe_port=50000
+      web_root=/var/www
+      ipxe_root=$web_root/$ipxe_app_name
+      tftp_root=/srv/tftp
+      external_iface=eno1
+      internal_iface=eno2
+      pxe_subnet=192.168.1
+      pxe_internal_ip=$pxe_subnet.1
+      pxe_subnet_mask_ip=255.255.255.0
+      pxe_subnet_bitmask=16
+
+#. Add the `pxe-server` bundle to your |CL| system. This bundle has all the
```

Comment:
Since all commands in this guide are entered as root, I would suggest that you state the the user needs to be logged in as root.

Reply:
Will do.